src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilerContext.cs

```diff
@@ -137,6 +137,22 @@ protected override RuntimeInterfacesAlgorithm GetRuntimeInterfacesAlgorithmForNo
         {
             return BaseTypeRuntimeInterfacesAlgorithm.Instance;
         }
+
+        TypeDesc _asyncStateMachineBox;
+        public TypeDesc AsyncStateMachineBoxType
+        {
+            get
+            {
+                if (_asyncStateMachineBox == null)
+                {
+                    _asyncStateMachineBox = SystemModule.GetType("System.Runtime.CompilerServices", "AsyncTaskMethodBuilder`1").GetNestedType("AsyncStateMachineBox`1");
```

Comment:
Nit: We have a GetKnownType and GetKnownNestedType extension method that throws "nicer" exceptions when things are missing.